src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -15,49 +15,109 @@
  */
 
 import {
-  Action,
-  ActionCreator,
   createSlice,
-  PayloadAction,
-  ThunkAction
+  type Action,
+  type ActionCreator,
+  type PayloadAction,
+  type ThunkAction
 } from '@reduxjs/toolkit';
 
 import browserStorageService from 'src/content/app/genome-browser/services/browserStorageService';
+import browserTrackConfigStorageService from 'src/content/app/genome-browser/components/browser-track-config/services/browser-track-config-storage-service';
 
 import { updateTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
 import { getBrowserTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+import { getGenomeTrackCategoriesById } from 'src/shared/state/genome/genomeSelectors';
+import { getActiveGenomeId } from 'src/content/app/species/state/general/speciesGeneralSelectors';
 
-import { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
-import { RootState } from 'src/store';
+import type { RootState } from 'src/store';
+import type { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import type { GenomeTrackCategory } from 'src/shared/state/genome/genomeTypes';
 
 export type CogList = {
   [key: string]: number;
 };
 
-export type TrackConfigState = Readonly<{
+export enum TrackType {
+  GENE = 'gene',
+  REGULAR = 'regular'
+}
+
+export const getTrackType = (trackId: string) => {
+  if (trackId.startsWith('gene')) {
+    return TrackType.GENE;
+  } else {
+    return TrackType.REGULAR;
+  }
+};
+
+export type GeneTrackConfig = {
+  showSeveralTranscripts: boolean;
+  showTranscriptIds: boolean;
+  showTrackName: boolean;
+  showFeatureLabel: boolean;
+  trackType: TrackType.GENE;
+};
+
+export type RegularTrackConfig = {
+  showTrackName: boolean;
+  trackType: TrackType.REGULAR;
+};
+
+export type TrackConfig = GeneTrackConfig | RegularTrackConfig;
+
+export type TrackConfigs = {
+  [trackId: string]: TrackConfig;
+};
+
+export type TrackConfigsForGenome = Readonly<{
   applyToAllConfig: {
-    isSelected: boolean;
-    allTrackNamesOn: boolean;
-    allTrackLabelsOn: boolean;
+    isSelected?: boolean;
   };
```

Comment:
Did we discuss whether it should be `applyToAll : boolean` rather than `applyToAllConfig: {...}`?